apps/test/unit/util/CachedElementTest.jsx

```diff
@@ -0,0 +1,139 @@
+import {expect} from '../../util/reconfiguredChai';
+import React from 'react';
+import {mount} from 'enzyme';
+import sinon from 'sinon';
+import CachedElement, {unitTestExports} from '@cdo/apps/util/CachedElement';
+
+const cache = unitTestExports.elementsHtmlCache;
+const defaultType = 'div';
+const defaultContent = '1';
+const defaultKeys = ['key1', 'key2', 'key3'];
+const defaultHtml = '<div>1</div>';
+const nestedHtml = `<div>${defaultHtml}</div>`;
+
+function getCacheSize(type = defaultType) {
+  return Object.keys(cache[type]).length;
+}
+
+function getCachedElement(type = defaultType, key = defaultKeys[0]) {
+  return cache[type][key];
+}
+
+function createDiv(content = defaultContent) {
+  return <div>{content}</div>;
+}
+
+function createNested() {
+  return <CachedElement {...defaultProps} />;
+}
+
+const defaultProps = {
+  elementType: defaultType,
+  cacheKey: defaultKeys[0],
+  createElement: createDiv
+};
+
+const spy = sinon.spy(createDiv);
+
+describe('CachedElement', () => {
+  afterEach(() => {
```

Comment:
(minor) Not sure if it makes a difference, but I like doing resets in beforeEach() to ensure that things in a particular state for the current test (as opposed to assuming that the previous test cleaned up properly).

(same comment for other test file)

Reply:
i went back and forth on this. it's also important that we clean up after ourselves, so i'll update this to `beforeEach` and add an `after` (called after this whole block of tests) that does the same.